salt/pillar/varstack_pillar.py

```diff
@@ -31,16 +29,14 @@
 except ImportError:
     pass
 
-# Set up logging
-log = logging.getLogger(__name__)
 
 # Define the module's virtual name
 __virtualname__ = 'varstack'
```

Comment:
You don't need that.

Reply:
The module does not use any logging at all (or I'm missing something).

Reply:
You do not need `__virtualname__` if the filename is the same. You use it only in case the module is one of those like zypper/apt/yum/whatever but we refer to it as `pkg`. Please remove this.

Reply:
The module filename is `varstack_pillar.py` and `__virtualname__ = 'varstack'`, so they are different.

Reply:
Srsly, I never understand why it is called that way. It is `salt.pillar.varstack` and `salt.module.varstack`. Why on Earth it needs to be called "Pizza pepperoni with meat pepperoni".

Reply:
I;ve just looked up. There are no imports, so it is safe.

Reply:
Also see https://github.com/saltstack/salt/blob/develop/salt/loader.py#L1789 (virtualname should match the returned value)

Reply:
@max-arnold this is in case when you do have it but misusing it:
https://github.com/saltstack/salt/blob/develop/salt/loader.py#L1770

Just remove it.

Reply:
This is when you have it:
https://github.com/saltstack/salt/blob/develop/salt/loader.py#L1770

But if you return just True and your filename is properly named, it is fine.

Reply:
If you have `$SALT/modules/example.py`:
```python
def __virtual__():
    return True, 'Some error message'

def hello():
    return 'hello'
```
Then:
```
salt-call --local example.hello -l error
local:
    hello
```

But:
```python
def __virtual__():
    return 'foo', 'Some error message'
```

...will render to this:

```
salt-call --local example.hello -l error
[ERROR   ] Failed to read the virtual function for module: example
```
And it won't even load the module.